specification/sbi_atee.adoc

```diff
@@ -13,6 +44,953 @@ running a VMM).
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | An input address was invalid.
+| SBI_ERR_INVALID_PARAM   | The input parameters were invalid, or the TVM wasn't
+                            in the correct state.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE TVM Add Shared Memory Region (FID #8)
+[source, C]
+-----
+struct sbiret sbi_tee_tvm_add_shared_memory_region(unsigned long tvm_guest_id,
+                                                   unsigned long tvm_gpa_addr,
+                                                   unsigned long region_len);
+-----
+Marks the range of TVM physical address space starting at `tvm_gpa_addr` as reserved
+for the mapping of shared memory. The region is initially unpopulated, and pages of
+shared memory may be inserted by calling `sbi_tee_tvm_add_shared_pages()`. Attempts
+by a TVM VCPU to access an unpopulated region will cause a `SHARED_PAGE_FAULT` exit
+from `sbi_tee_tvm_run_vcpu()`. Both `tvm_gpa_addr` and `region_len` must be 4kB-aligned.
+
+This call may not not be made after calling `sbi_tee_tvm_finalize()`.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tvm_add_shared_memory_region_errors]
+.TEE TVM Add Shared Memory Region
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | An input address was invalid.
+| SBI_ERR_INVALID_PARAM   | The input parameters were invalid, or the TVM wasn't
+                            in the correct state.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE TVM Add Measured Pages (FID #9)
+[source, C]
+-----
+struct sbiret sbi_tee_tvm_add_measured_pages(unsigned long tvm_guest_id,
+                                             unsigned long source_address,
+                                             unsigned long dest_address,
+                                             unsigned long tsm_page_type,
+                                             unsigned long num_pages,
+                                             unsigned long tvm_guest_gpa);
+
+-----
+Copies num_pages pages from non-confidential memory at `source_address` to confidential
+memory at `dest_addr`, then measures and maps the pages at `dest_addr` at the TVM physical
+address space at `tvm_guest_gpa. The mapping must lie within a region of confidential memory
+created with `sbi_tee_tvm_add_confidential_memory_region()`. The tsm_page_type parameter must
+be a legal value for enum type `tsm_page_type`.
+
+This call may not not be made after calling `sbi_tee_tvm_finalize()`.
+
+The possible error codes returned in `sbiret.error` are shown below.
```

Comment:
I wonder if we should have an error code for "not enough page table pages" for the various `add_*_pages` calls.